src/exec.js

```diff
@@ -13,105 +11,44 @@ common.register('exec', _exec, {
-  if (fs.existsSync(stdoutFile)) common.unlinkSync(stdoutFile);
-  if (fs.existsSync(stderrFile)) common.unlinkSync(stderrFile);
-  if (fs.existsSync(codeFile)) common.unlinkSync(codeFile);
+  opts.encoding = 'utf8';
 
-  var execCommand = JSON.stringify(common.config.execPath) + ' ' + JSON.stringify(scriptFile);
-  var script;
+  if (pipe) {
+    opts.input = pipe;
+  }
 
   opts.cwd = path.resolve(opts.cwd);
-  var optString = JSON.stringify(opts);
-
-  script = [
-    "var child = require('child_process')",
-    "  , fs = require('fs');",
-    'var childProcess = child.exec(' + JSON.stringify(cmd) + ', ' + optString + ', function(err) {',
-    '  var fname = ' + JSON.stringify(codeFile) + ';',
-    '  if (!err) {',
-    '    fs.writeFileSync(fname, "0");',
-    '  } else if (err.code === undefined) {',
-    '    fs.writeFileSync(fname, "1");',
-    '  } else {',
-    '    fs.writeFileSync(fname, err.code.toString());',
-    '  }',
-    '});',
-    'var stdoutStream = fs.createWriteStream(' + JSON.stringify(stdoutFile) + ');',
-    'var stderrStream = fs.createWriteStream(' + JSON.stringify(stderrFile) + ');',
-    'childProcess.stdout.pipe(stdoutStream, {end: false});',
-    'childProcess.stderr.pipe(stderrStream, {end: false});',
-    'childProcess.stdout.pipe(process.stdout);',
-    'childProcess.stderr.pipe(process.stderr);',
-  ].join('\n') +
-    (pipe ? '\nchildProcess.stdin.end(' + JSON.stringify(pipe) + ');\n' : '\n') +
-    [
-      'var stdoutEnded = false, stderrEnded = false;',
-      'function tryClosingStdout(){ if(stdoutEnded){ stdoutStream.end(); } }',
-      'function tryClosingStderr(){ if(stderrEnded){ stderrStream.end(); } }',
-      "childProcess.stdout.on('end', function(){ stdoutEnded = true; tryClosingStdout(); });",
-      "childProcess.stderr.on('end', function(){ stderrEnded = true; tryClosingStderr(); });",
-    ].join('\n');
-
-  fs.writeFileSync(scriptFile, script);
 
   /* istanbul ignore else */
-  if (opts.silent) {
-    opts.stdio = 'ignore';
+  if (silent) {
+    opts.stdio = 'pipe';
   } else {
-    opts.stdio = [0, 1, 2];
+    opts.stdio = 'inherit';
   }
 
-  // Welcome to the future
-  try {
-    child.execSync(execCommand, opts);
-  } catch (e) {
-    // Clean up immediately if we have an exception
-    try { common.unlinkSync(scriptFile); } catch (e2) {}
-    try { common.unlinkSync(stdoutFile); } catch (e2) {}
-    try { common.unlinkSync(stderrFile); } catch (e2) {}
-    try { common.unlinkSync(codeFile); } catch (e2) {}
-    throw e;
-  }
-
-  // At this point codeFile exists, but it's not necessarily flushed yet.
-  // Keep reading it until it is.
-  var code = parseInt('', 10);
-  while (isNaN(code)) {
-    code = parseInt(fs.readFileSync(codeFile, 'utf8'), 10);
-  }
+  var c = child.spawnSync(cmd, opts);
 
-  var stdout = fs.readFileSync(stdoutFile, 'utf8');
-  var stderr = fs.readFileSync(stderrFile, 'utf8');
+  stdout = c.stdout || '';
+  stderr = c.stderr || '';
```

Comment:
Thanks, you're right. I didn't realize this. It's fixed now.

Btw. I noticed strange behavior. If you run an interactive command with `exec`, eg. `git commit` (with `vim` configured as an `EDITOR`) previous version of `execSync` will end up in not so functional state, you have to basically end up the script with Ctrl+C.

The current version just print out something like this:
```
Vim: Warning: Output is not to a terminal
Vim: Warning: Input is not from a terminal
Vim: Error reading input, exiting...

Vim: Finished.
error: There was a problem with the editor 'nvim'.
Please supply the message using either -m or -F option.
```